src/Dialect/ONNX/ONNXShapeHelper.hpp

```diff
@@ -63,6 +63,41 @@ struct ONNXOpShapeHelper {
   SmallVector<DimsExpr, 1> outputsDims;
 };
 
+/// Compute a broadcasted shape from the shapes of given operands. Operands must
+/// be ranked in advance.
+struct ONNXOpBroadcastedShapeHelper {
```

Comment:
My first reaction would be to have this as a subclass of the main ShapeHelper class; thinking a bit more, maybe it does not really have to, because I don't really think we will use them in a generic way. So its probably fine as is.